File: 백준/Silver/1764. 듣보잡/듣보잡.cc
#include <bits/stdc++.h>

using namespace std;

set<string> s;
vector<string> res;
int main() {
	int n, m;
	ios::sync_with_stdio(0);
	cin.tie(0);

	cin >> n >> m;
	
	for (int i =0 ;i < n; i++)  {
		string temp;
		cin >> temp;
		s.insert(temp);
	}
	for (int i = 0; i< m; i++) {
		string temp;
		cin >> temp;
		if (s.count(temp)) {
			res.push_back(temp);
		}
	}
	cout << res.size() << '\n';
	sort(res.begin(), res.end());
	for (auto i : res) {
		cout << i << '\n';
	}
}

File: 백준/Silver/1764. 듣보잡/듣보잡.py

n, k = map(int, input().split())

dict = {}
res = []
for i in range(n):
    t = str(input())
    dict[t] = 1
for i in range(k):
    t = str(input())
    if t in dict.keys():
        res.append(t)

print(len(res))
res.sort()
for i in res:
    print(i)